modules/apps/object/object-dynamic-data-mapping/src/main/resources/META-INF/resources/Autocomplete.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import {debounce} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+function getData(apiUrl, query, page, pageSize) {
+	const url = new URL(apiUrl, Liferay.ThemeDisplay.getPortalURL());
+
+	if (query) {
+		url.searchParams.append('search', query);
+	}
+
+	if (page) {
+		url.searchParams.append('page', page);
+	}
+
+	if (pageSize) {
+		url.searchParams.append('pageSize', pageSize);
+	}
+
+	return Liferay.Util.fetch(url, {
```

Comment:
You can import it from `frontend-js-web` together with `{debounce}`. It would be great if, together with `[no-global-fetch](https://github.com/liferay/liferay-frontend-projects/blob/master/projects/eslint-config/plugins/portal/lib/rules/no-global-fetch.js)` we could make sure that we use the imported one when in a module context